src/mono/mono/mini/intrinsics.c

```diff
@@ -1763,7 +1763,7 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 					result = cosh (source);
 					break;
 				case OP_ROUND:
-					result = round (source);
+					result = round (source * 0.5) * 2.0;
```

Comment:
I'm confused.   Doesn't this give the wrong answer for odd whole numbers? 

if `source = 3.0` then `result = round (3.0 * 0.5) * 2.0` = `round (1.5) * 2.0` = (midpoints round away from zero) = `2.0 * 2.0` = `4.0`

Reply:
Oops, yeah, copied the wrong fix 🙂